.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,11 @@
+name: run pre-commit hooks
+
+on: [push]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
```

Comment:
`actions/setup-python@v4`

Reply:
What's the difference?

I copy / pasted it from [here](https://github.com/pre-commit/action). They seam to use v3. Maybe its necessary?

Reply:
I think it will not change anything, but dependabot might annoy us if we use this "outdated" version of the action. The other workflows use `@v4`.

Reply:
ok, i update to v4